crates/ruff_python_semantic/src/model.rs

```diff
@@ -662,12 +662,117 @@ impl<'a> SemanticModel<'a> {
                 }
             }
 
+            // FIXME: Shouldn't this happen above where `class_variables_visible` is set?
```

Comment:
Why? Do you plan to follow-up on this? It seems unrelated to the PR.

Reply:
Because for `BindingKind::Annotation` this assignment is skipped. Maybe there's something subtle at play here, like `BindingKind::Annotation` never occurring inside a function scope, but otherwise this seems like a bug, albeit one very unlikely to affect anyone in any real way.

I noted it because `simulate_runtime_load` has a similar structure but moved the update of this variable above to where `class_variable_visible` is updated.

If you don't think there's a bug here I'm happy to remove the comment, otherwise I'll either leave it in and follow up or fix it as part of this PR, whichever you'd prefer.

Reply:
Let's address this in a follow up PR